### PR TITLE
[Product Data] First step in gateway usage data collection

### PR DESCRIPTION
**Pull request by simonwicky**
## Overview 
This PR is the first step towards collecting data on gateway usage. It builds up on @neacsu's old code for what was then nym-connect.

## What it does
Exposes unique users count and connection time histogram on the `metrics/sessions` endpoint of the self-described API on entry-gateways. For the time being, data is collected by probing the `ActiveClientStore` every minute and extracting data from this. Data is stored internally and exposed on the next day, i.e. UTC day $d$ exposes data from day $d-1$.

## Future plans 
The [`statistics`](https://github.com/nymtech/nym/tree/simon/product_data/gateway/src/node/statistics) module will evolve as we add collection for product data and censorship resistance study. The collection will also eventually switch from probing to event-based for more accurate data.

- - -
This change is [Reviewable](https://reviewable.io/reviews/nymtech/nym/4963)

### CODE REVIEW

File: gateway/src/node/mod.rs [hunk 1]
```diff
@@ -636,6 +660,13 @@ impl<St> Gateway<St> {
             shutdown.fork("mixnet_handling::Listener"),
         );
 
+        let shared_session_stats = self.session_stats.take().unwrap_or_default();
```
Comment (octol) [inline gateway/src/node/mod.rs]: Maybe just skip starting it if it has not been set? Or maybe it is assumed to always be running?

File: gateway/src/node/mod.rs [hunk 2]
```diff
@@ -636,6 +660,13 @@ impl<St> Gateway<St> {
...
+        self.start_stats_collector(
+            active_clients_store.clone(),
+            shared_session_stats,
+            shutdown.fork("statistics::sessionCollector"),
```
Comment (octol) [inline gateway/src/node/mod.rs]: `camelCase` makes me think of Java and immediately get chills down my spine. Let's use `PascalCase` ;)
Also, isn't it called `StatisticsCollector` or `GatewayStatisticsCollector`?

**Review by octol — APPROVED** [review]

File: gateway/src/node/mod.rs (hunk 2 above)
Comment (simonwicky) [reply gateway/src/node/mod.rs]: Sorry for the chills :)

Fixed in 65b83627fc31506ef8725adf0cd21632f68a55aa

File: gateway/src/node/mod.rs (hunk 1 above)
Comment (simonwicky) [reply gateway/src/node/mod.rs]: I got a bit baited by that similar construction in [`mixing_stats`](https://github.com/nymtech/nym/blob/08aa0af56242bdc2fcf46f5e07b05557855f61e5/mixnode/src/node/mod.rs#L133).
If we have to create a default one here it won't be passed to the self described API so it's not really useful to start it.

Skipping start in 65b83627fc31506ef8725adf0cd21632f68a55aa

**Review by octol — APPROVED** [review]

**Review by mmsinclair — APPROVED** [review]

**Pull request merged**